packages/suite/src/actions/suite/autoForgetDeviceDataThunks.ts

```
@@ -26,6 +26,7 @@ export const setAutoForgetDeviceDataThunk = createThunk<
     // Finally, having purged the data in Bluetooth and THP reducers, simply sync BT and THP to persistent storage.
     await dispatch(storageActions.saveKnownDevices());
     await dispatch(storageActions.saveThpCredentials());
+    storageActions.clearPersistentDeviceData();
```

no await needed?

---

No, it's a side effect that runs async. As you can see, the current actions like `saveKnownDevices` or `saveThpCredentials` all do. So the await really just awaits redux, not the inner thunk function body.
As I think about it, the awaits are quite useless..

---

lol, i see

---

we just talked about this right? :D

---

hmm not really.. This is just a redux specific thing, that it handles dispatching of thunks always async, even if the thunk inner function is sync.

---

oh wait a moment, the original `savePersistentDeviceData` thunk **should have been async** :scream: 
Both `saveKnownDevices` and `saveThpCredentials` are already async :heavy_check_mark: 
Thanks for the catch! :bow:  Fixed.

All `db` operations are async operations, and if I do several of them in sequence, I have to await them, or `db` will be unavailable, and it will silently fail because of that condition.
This one was the last, so not awaiting it would have worked incidentally :sweat_smile: 
But for example `saveKnownDevices` and `saveThpCredentials` would break if I made them unawaitable...